### PR TITLE
[Fix] fix topdown demo bug when no obj detected

### PR DESCRIPTION
**Pull request by Tau-J**
## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: demo/topdown_demo_with_mmdet.py [hunk 1]
```diff
@@ -172,7 +175,11 @@ def main():
             pose_estimator,
             visualizer,
             show_interval=0)
-        pred_instances_list = split_instances(pred_instances)
+
+        if pred_instances is not None:
+            pred_instances_list = split_instances(pred_instances)
+        else:
+            pred_instances_list = []
```
Comment (Ben-Louis) [inline demo/topdown_demo_with_mmdet.py:182]: Would it be better if we improved the `split_instances` function to handle `None` inputs and return an empty list?

Comment (Tau-J) [reply demo/topdown_demo_with_mmdet.py:182]: good idea

File: demo/topdown_demo_with_mmdet.py [hunk 2]
```diff
@@ -60,7 +60,10 @@ def process_one_image(args, img_path, detector, pose_estimator, visualizer,
         out_file=out_file,
         kpt_score_thr=args.kpt_thr)
 
-    return data_samples.pred_instances
+    if hasattr(data_samples, 'pred_instances'):
+        return data_samples.pred_instances
+    else:
+        return None
```
Comment (ly015) [inline demo/topdown_demo_with_mmdet.py:66]:
```suggestion
    return data_samples.get('pred_instances', None)
```

File: mmpose/structures/utils.py [hunk 3]
```diff
@@ -120,6 +120,9 @@ def split_instances(instances: InstanceData) -> List[InstanceData]:
     information about one instance."""
     results = []
 
+    if instances is None:
```
Comment (ly015) [inline mmpose/structures/utils.py:123]: Please add a few comments to explain this is the case where no object is detected.

**Pull request merged**